api/src/main/java/org/openmrs/module/drawing/elements/DrawingSubmissionElement.java

```diff
@@ -410,7 +411,7 @@ public String generateHtml(FormEntryContext context) {
 					
 				}
 				
-			} else if (ctxMode == Mode.ENTER) {
+			} else if (ctxMode == Mode.ENTER || ( parentObs != null && parentObs.getComplexData().getData() == null) ) {
```

Comment:
I thought that the reported error happens only in edit mode.

Reply:
I think it also happens on view mode.

Reply:
@icrc-jfigueiredo can you confirm? Does it also happen when submitting a new form?

Reply:
When submitting a new form it does not happen, because the main problem is related to a saved file, and when we are in enter mode we don't have a file yet.

That part of the code is only saying if we are on "Enter Mode" or if the drawing file does not open, we will get an empty drawing.

Reply:
> And we can confirm that it also happens on View Mode.
> But in Enter mode it does not happen, because the main problem is related to a saved file, and when we are in enter mode we don't have a file yet.
> 
> That part of the code is only saying if we are on "Enter Mode" or if the drawing file does not open, we will get an empty drawing.

@dkayiwa  does this pr look good enough now to you!

Reply:
Because when we are on "Enter Mode" we add a new empty drawing, and on this case (that the file is missing) we want the same thing, so i add an or condition `|| ( parentObs != null && parentObs.getComplexData().getData() == null)`

Reply:
We can have another else if, with the same code that we have on the enter mode